ggshield/cmd/iac/scan/prereceive.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import os
+import sys
+from pathlib import Path
+from typing import Any, Optional, Sequence, Tuple
+
+import click
+
+from ggshield.cmd.iac.scan.diff import iac_scan_diff
+from ggshield.cmd.iac.scan.iac_scan_common_options import (
+    add_iac_scan_common_options,
+    update_context,
+)
+from ggshield.cmd.iac.scan.iac_scan_utils import (
+    IaCSkipScanResult,
+    create_output_handler,
+)
+from ggshield.core.errors import UnexpectedError
+from ggshield.core.git_shell import git
+from ggshield.core.text_utils import display_error
+from ggshield.core.utils import EMPTY_SHA, EMPTY_TREE, PRERECEIVE_TIMEOUT
+from ggshield.iac.collection.iac_diff_scan_collection import IaCDiffScanCollection
+
+
+logger = logging.getLogger(__name__)
+
+REMEDIATION_MESSAGE = """  A pre-receive hook set server side prevented you from pushing secrets.
+  Since the secret was detected during the push BUT after the commit, you need to:
+  1. rewrite the git history making sure to replace the secret with its reference (e.g. environment variable).
+  2. push again."""
+
+BYPASS_MESSAGE = """\n     git push -o breakglass"""
+
+
+def get_prereceive_timeout() -> float:
+    try:
+        return float(os.getenv("GITGUARDIAN_TIMEOUT", PRERECEIVE_TIMEOUT))
+    except BaseException as e:
+        display_error(f"Unable to parse GITGUARDIAN_TIMEOUT: {str(e)}")
+        return PRERECEIVE_TIMEOUT
+
+
+def get_breakglass_option() -> bool:
+    """Test all options passed to git for `breakglass`"""
+    raw_option_count = os.getenv("GIT_PUSH_OPTION_COUNT", None)
+    if raw_option_count is not None:
+        option_count = int(raw_option_count)
```

Comment:
Since you catch errors related to parsing `GITGUARDIAN_TIMEOUT`, shouldn't you do the same for `GIT_PUSH_OPTION_COUNT`? (in case it's not a correct `int`)

Reply:
This comes from the secrets side for pre-receive, and will be refactored / corrected.